app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+class CommentsController < ApplicationController
+
+  def index
+    @comments= Comment.all.order(:id)
+  end
+
+  def new
+    @comment= Comment.new
+  end
+
+  def create
+    @comment = Comment.create!(comment_params)
+    redirect_to "/posts/:id"
```

Comment:
This redirect statement won't take you where you wanna go. Instead, it's literally going to take you to "/posts/:id" (where :id is the string ":id").

Instead try `redirect_to post_path(@post)`. This assumes you have defined `@post`